test/hotspot/jtreg/vmTestbase/vm/mlvm/meth/share/MHTransformationGen.java

```diff
@@ -88,6 +102,12 @@ public static MHMacroTF createSequence(Argument finalRetVal, Object boundObj, Me
         List<MHTFPair> pendingPWTFs = new LinkedList<MHTFPair>();
 
         for ( int i = nextInt(MAX_CYCLES); i > 0; i-- ) {
+            if (isCodeCacheEffectivelyFull()) {
+                Env.traceNormal("Not enought code cache to build up MH sequences anymore. " +
+                        " Has only been able to achieve " + (MAX_CYCLES - i) + " out of " + MAX_CYCLES);
```

Comment:
given `nextInt(x)` returns a random number from `[0; x]`, we might have achieved more (or less) `MAX_CYCLES - i`, i.e. that part of the message is incorrect, I'd just remove it.

Reply:
Fixed by extracting the generated random number first.